### ISSUE TITLE
Some members are being charged twice for their subscription (Stripe)

### ISSUE THREAD

**Issue by thomasfw** [restrictcontentpro/restrict-content-pro#1393]
**We're still on 2.8.6 (were waiting on 2.9.1 before upgrading which I see has now been released). Is this a known issue and was there a fix in 2.9?**

See the membership logs below for a couple of affected users. From what I can tell, it looks like this is only happening when a user's recurring payment fails. Presumably they've gone in to renew manually and this has triggered a new payment + authorised the original renewal payment on top. Although i'm still looking through the records to see just how many users this has affected.

If there's already a fix for this and I upgrade, do I need to change anything else in Stripe manually or is it okay to just refund the additional payments. (i.e. This won't have set up an extra subscription for those affected users or anything like that?).

```
August 24, 2016 20:37:58 - Member's status changed from  to pending

August 24, 2016 20:38:05 - Member's status changed from pending to active

August 24, 2016 20:38:23 - Member's expiration changed from  to 2017-08-24 23:59:59

August 24, 2017 21:40:03 - Recurring charge failed in Stripe. Event ID: evt_****

August 27, 2017 02:07:19 - Member's status changed from active to expired

August 27, 2017 21:40:19 - Recurring charge failed in Stripe. Event ID: evt_****

August 28, 2017 13:02:07 - Associate Membership subscription renewed.

August 28, 2017 13:02:07 - Member's status changed from expired to pending

August 28, 2017 13:02:15 - Member's expiration changed from 2017-08-24 23:59:59 to 2018-08-28 23:59:59

August 28, 2017 13:02:16 - Member's status changed from pending to active

August 28, 2017 13:02:21 - Member's expiration changed from 2018-08-28 23:59:59 to 2018-08-24 23:59:59

August 28, 2017 13:03:21 - Member's expiration changed from 2018-08-24 23:59:59 to 2018-08-28 23:59:59
```

```
July 2, 2016 02:03:22 - The subscription was set up with the batch user import.

August 25, 2016 14:36:16 - Member's status changed from free to pending

August 25, 2016 14:36:25 - Member's status changed from pending to active

August 25, 2016 14:36:34 - Member's expiration changed from none to 2017-08-25 23:59:59

August 25, 2017 15:39:07 - Recurring charge failed in Stripe. Event ID: evt_***

August 28, 2017 02:06:39 - Member's status changed from active to expired

August 28, 2017 15:40:38 - Recurring charge failed in Stripe. Event ID: evt_***

August 28, 2017 16:13:32 - Regular Membership subscription renewed.

August 28, 2017 16:13:33 - Member's status changed from expired to pending

August 28, 2017 16:13:40 - Member's expiration changed from 2017-08-25 23:59:59 to 2018-08-28 23:59:59

August 28, 2017 16:13:41 - Member's status changed from pending to active

August 28, 2017 16:13:44 - Member's expiration changed from 2018-08-28 23:59:59 to 2018-08-25 23:59:59

August 28, 2017 16:13:45 - Member's expiration changed from 2018-08-25 23:59:59 to 2018-08-28 23:59:59
```

**Comment by thomasfw** [comment]
Those members also now have duplicated subscriptions in Stripe.. Looks like i've got some explaining and a few days worth of manual searches and reconciliation to do :'(

<img width="1118" alt="screen shot 2017-08-30 at 22 09 20" src="https://user-images.githubusercontent.com/7723105/29892741-1a7572e8-8dd0-11e7-8c0b-c49c0b1d64ed.png">

**Comment by mindctrl** [comment]
This sounds like a duplicate of https://github.com/restrictcontentpro/restrict-content-pro/issues/962.

**Comment by thomasfw** [comment]
Can you suggest some steps to remedy this until a fix is released, it's kind of a big deal

**Comment by thomasfw** [comment]
@pippinsplugins what's the status on this, it's an absolute nightmare for us. I see the other issue was posted around 7 months ago?

**Comment by mindctrl** [comment]
Closing this out since it's a duplicate.

@thomasfw the status can be seen on #962. If you want to test the suggested patch there, please do. We'll be doing additional testing on it soon.
